ftlib/commlib/pytorch/trick.py

```diff
@@ -0,0 +1,75 @@
+import itertools
+
+import torch
+
+
+class TrickyIterator:
```

Comment:
Why is this tricky? We may want to add some doc strings to illustrate

Reply:
Here comes a naming issue again. When I name it after `Tricky`, I was implying that this iterator is not like the built-in one with consistent iterative content. Such `Tricky` Sampler/Iterator shall be aware of changes of `rank` and `size` from `FTLib`.

Would you suggest a new name or let me put more description to the class definitions?

Reply:
Since it’s just a demo, it should be fine for now. We can revisit later